### DISCUSSION TITLE
Proposal: Allow the name of an out variable to be omitted if it is not used

### DISCUSSION THREAD

**Discussion by jmairboeck** [dotnet/csharplang#1125] [General]
It should be allowed to omit the variable name of a variable declared in an out expression, if the variable is not going to be used (similar to the catch clause for example), like this:

```c#
string str = ...
var isValidInt = int.TryParse(str, out int); // or out var
// the actual value is not needed here
```

Currently, when just saying `out int`, it is considered an "invalid expression". In the `out var` case, it is reported as "identifier var not found" (because var is a contextual keyword).

**Comment by CyrusNajmabadi** [comment]
Use ```_```.  https://docs.microsoft.com/en-us/dotnet/csharp/discards

```c#
string str = ...
var isValidInt = int.TryParse(str, out _);
```
